src/Dict.h

```diff
@@ -138,6 +150,76 @@ class DictEntry {
 
 } // namespace detail
 
+class DictIterator {
+public:
+	using value_type = detail::DictEntry;
+	using reference = detail::DictEntry&;
+	using pointer = detail::DictEntry*;
+	using difference_type = std::ptrdiff_t;
+	using iterator_category = std::forward_iterator_tag;
+
+	~DictIterator();
+
+	reference operator*() { return *curr; }
+	pointer operator->() { return curr; }
+
+	DictIterator& operator++();
+	DictIterator operator++(int) { auto temp(*this); ++*this; return temp; }
+
+	bool operator==( const DictIterator& that ) const { return curr == that.curr; }
+	bool operator!=( const DictIterator& that ) const { return !(*this == that); }
+
+private:
+	friend class Dictionary;
+
+	DictIterator() = default;
+	DictIterator(const Dictionary* d, detail::DictEntry* begin, detail::DictEntry* end);
+
+	Dictionary* dict = nullptr;
+	detail::DictEntry* curr = nullptr;
+	detail::DictEntry* end = nullptr;
+};
+
+class RobustDictIterator {
+public:
+	using value_type = detail::DictEntry;
+	using reference = detail::DictEntry&;
+	using pointer = detail::DictEntry*;
+	using difference_type = std::ptrdiff_t;
+	using iterator_category = std::forward_iterator_tag;
+
+	RobustDictIterator() : curr(nullptr) {}
+	RobustDictIterator(Dictionary* d);
+	RobustDictIterator(const RobustDictIterator& other);
+	RobustDictIterator(RobustDictIterator&& other);
+ 	~RobustDictIterator();
+
+	reference operator*() { return curr; }
+	pointer operator->() { return &curr; }
+
+	RobustDictIterator& operator++();
+//	RobustDictIterator operator++(int) { auto temp(*this); ++*this; return temp; }
```

Comment:
Going to uncomment this in the merge since it looks like what I expect, but just flagging in case I lost track of a reason for it.